src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -209,7 +209,7 @@ public void OnStopActivity(Activity activity, object payload)
 
                 if (this.stopResponseFetcher.TryFetch(payload, out HttpResponseMessage response) && response != null)
                 {
-                    activity.SetTag(SemanticConventions.AttributeHttpStatusCode, (int)response.StatusCode);
+                    activity.SetTag(SemanticConventions.AttributeHttpStatusCode, TelemetryHelper.GetBoxedStatusCode(response.StatusCode));
```

Comment:
nit: Call this in the instrumentation ctor to invoke the static `TelemetryHelper` ctor to populate the array?

Reply:
Updated on https://github.com/open-telemetry/opentelemetry-dotnet/pull/4363